src/moped/src/main/java/core/sensors/NoiseNumberGenerator.java

```diff
@@ -0,0 +1,171 @@
+     * @param value The default value.
+     * @return The altered number.
+     */
+    public double generateNoisedValue(double value) {
+        return noisedNumber(value, maxNormalDelta, randomSpikeRate, spikeModifier);
+    }
+
+    /**
+     * Gives a number some random noise and random chance of spiked value.
+     * @param value The default value.
+     * @param maxDelta Max Noise allowed.
+     * @param spikeRate Chance modifier for a spike
+     * @param spikeValueModifier Modifier value for a spike chance.
+     * @return The altered number.
+     */
+    private double noisedNumber(double value, double maxDelta, double spikeRate, double spikeValueModifier) {
+        double noiseValue;
+        if (rand.nextDouble() <= spikeRate) {
+            noiseValue = (rand.nextDouble()*spikeValueModifier * maxDelta) - maxDelta;
+        } else {
+            noiseValue = (rand.nextDouble()*2 * maxDelta) - maxDelta;
+        }
+        return value + noiseValue;
+    }
+
+    /**
+     * Creates an array of values where the first half are rising and the lower half is falling,
+     * with all values affected of random noise and spikes.
+     * @param lowValues Lowest value (without added noise) in the array.
+     * @param highValues Highest value (without added noise)
+     * @param size Size of the array.
+     * @return The array that contains all of the values.
+     */
+    public double[] generateNormalNoiseCurve(double lowValues, double highValues, int size) {
+        double[] noiseCurve = new double[size];
+
+        // Rising half
+        for (int i = 0; i < size / 2; i++) {
+            double normalValue = lowValues + (highValues - lowValues) * (i * (1/((double) size/2)));
+            noiseCurve[i] = noisedNumber(normalValue, maxNormalDelta, randomSpikeRate, spikeModifier);
+        }
+
+        // Falling half
+        for (int i = 0; i < size / 2; i++) {
+            double normalValue = highValues - (highValues - lowValues) * (i * (1/((double) size/2)));
+            noiseCurve[size / 2 + i] = noisedNumber(normalValue, maxNormalDelta, randomSpikeRate, spikeModifier);
+        }
+
+        return noiseCurve;
+    }
+
+    public double[] generateNoisySinusCurve(double lowestValue, double highestValue, int sinusPeriod, int size) {
+        double[] noisyCurve = new double[size];
+
+        for (int i = 0; i < size; i++) {
+            double normalValue = ((highestValue - lowestValue)/2) * Math.sin((2*Math.PI / sinusPeriod) * i) + (lowestValue + highestValue) / 2;
+            double noisyValue = generateNoisedValue(normalValue);
+            noisyCurve[i] = noisyValue;
+        }
+
+        return noisyCurve;
+    }
+
+    /**
+     * Uses {@Link LowPassFilter} to print out all of the values created by different
+     * weight values in a LowPassFilter, to the console. The data is based on the rawValues parameter
+     * that the LowPassFilter will use.
+     *
```

Comment:
A bit too much information maybe? The user will see the output when the method is in use :)